### DISCUSSION TITLE
Syntax Highlighting For Pug

### DISCUSSION THREAD

**Discussion by wgmyers** [geany/geany#4052] [Show and tell]
I am very new to Geany, and found that it did not support syntax highlighting for Pug, an HTML templating language I use in a number of projects.

Unable to find an existing project for Pug support in Geany, I have implemented a very rough and ready version and released it here:

https://github.com/wgmyers/geany-pug-support

Pug syntax is a bit odd and not really like anything obvious like C, HTML, Markdown etc, so I think that do to it properly would involve writing a lexer in C++, which is currently beyond my abilities. For now, this avoids having no syntax highlighting at all.

I hope this is helpful to someone (and if there was an existing implementation that I failed to find, my apologies, and please let me know where to find it, as it is likely better than mine is...)

**Comment by eht16** [comment]
If you like you can transform your filetype config to the Geany wiki, so more users can find and use it. E.g. create a new page under https://wiki.geany.org/config/start.

For the Scintilla lexer for syntax highlighting, you could add a feature request to the Scintilla/Lexilla project, maybe someone is willing to write a Pug lexer.
https://github.com/ScintillaOrg/lexilla/issues

I'm closing this as it is unlikely that we will include the config into Geany directly.
